libraries/botbuilder/tests/cloudAdapter.test.js

```diff
@@ -87,6 +98,59 @@ describe('CloudAdapter', function () {
 
             mock.verify();
         });
+
+        it('throws exception on expired token', async function () {
+            const consoleSpy = sinon.spy(console, 'error');
+            // Expired token with removed AppID
+            const authorization =
+                'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6IjJaUXBKM1VwYmpBWVhZR2FYRUpsOGxWMFRPSSIsImtpZCI6IjJaUXBKM1VwYmpBWVhZR2FYRUpsOGxWMFRPSSJ9.eyJhdWQiOiJodHRwczovL2FwaS5ib3RmcmFtZXdvcmsuY29tIiwiaXNzIjoiaHR0cHM6Ly9zdHMud2luZG93cy5uZXQvZDZkNDk0MjAtZjM5Yi00ZGY3LWExZGMtZDU5YTkzNTg3MWRiLyIsImlhdCI6MTY3MDM1MDQxNSwibmJmIjoxNjcwMzUwNDE1LCJleHAiOjE2NzA0MzcxMTUsImFpbyI6IkUyWmdZTkJONEpWZmxlOTJUc2wxYjhtOHBjOWpBQT09IiwiYXBwaWQiOiI5ZGRmM2QwZS02ZDRlLTQ2MWEtYjM4Yi0zMTYzZWQ3Yjg1NmIiLCJhcHBpZGFjciI6IjEiLCJpZHAiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC9kNmQ0OTQyMC1mMzliLTRkZjctYTFkYy1kNTlhOTM1ODcxZGIvIiwicmgiOiIwLkFXNEFJSlRVMXB2ejkwMmgzTldhazFoeDIwSXpMWTBwejFsSmxYY09EcS05RnJ4dUFBQS4iLCJ0aWQiOiJkNmQ0OTQyMC1mMzliLTRkZjctYTFkYy1kNTlhOTM1ODcxZGIiLCJ1dGkiOiJIWDlncld2bU1rMlhESTRkS3BHSEFBIiwidmVyIjoiMS4wIn0.PBLuja5sCcDfFjweoy-VucvbfHEyEcs1GyqXjekzBqgvK-mSc1UrEfqr5834qY6dLNsXVIMJzMFuH6WyPbnAfIfRcabdiVSOAl8N8e9Tex6vHfPi4h4P2F96VkXU80EtZX4QMjsJMDJ5eXbJlIDEAxXoJbAdHqgy-lHcVBx8XK7toJ_W7vSsFhis3C4CPCHI1cf1WuHVwfFXBiNwsOzj9cnRUKpea6UELV89q4C0L6aeSNdWYXehZmgq-wlo2wIaGgQ7rOXx4MlIrc83LBzMMc6TWvBJecK6O8pJWLe6BTwOltBI8Tmo2hWnY1OnsbOhbSSlfwLaZqKI7QpA50_2GQ';
+
+            const activity = { type: ActivityTypes.Invoke, value: 'invoke' };
+
+            const req = httpMocks.createRequest({
+                method: 'POST',
+                headers: { authorization },
+                body: activity,
+            });
+
+            const res = httpMocks.createResponse();
+
+            const logic = async (context) => {
+                context.turnState.set(INVOKE_RESPONSE_KEY, {
+                    type: ActivityTypes.InvokeResponse,
+                    value: {
+                        status: 200,
+                        body: 'invokeResponse',
+                    },
+                });
+            };
+
+            const validTokenIssuers = [];
+            const claimsValidators = allowedCallersClaimsValidator(['*']);
+            const authConfig = new AuthenticationConfiguration([], claimsValidators, validTokenIssuers);
+            const credentialsFactory = new ConfigurationServiceClientCredentialFactory({
+                MicrosoftAppId: '',
+                MicrosoftAppPassword: '',
+                MicrosoftAppType: '',
+                MicrosoftAppTenantId: '',
+            });
+
+            const botFrameworkAuthentication = createBotFrameworkAuthenticationFromConfiguration(
```

Comment:
Is the authentication required?, in dotnet for example if CouldAdapter doesn't receive it, it will create an Anonymous one, if we require this, dismiss this

Reply:
Discarded as it's related to changes in another PR.